azure-iot-device/azure/iot/device/provisioning/internal/polling_machine.py

```diff
@@ -0,0 +1,325 @@
+
+        states = [
+            "disconnected",
+            "initializing",
+            "registering",
+            "waiting_to_poll",
+            "polling",
+            "completed",
+            "error",
+            "cancelling",
+        ]
+
+        transitions = [
+            {
+                "trigger": "_trig_register",
+                "source": "disconnected",
+                "before": "_initialize_register",
+                "dest": "initializing",
+            },
+            {
+                "trigger": "_trig_register",
+                "source": "error",
+                "before": "_initialize_register",
+                "dest": "initializing",
+            },
+            {"trigger": "_trig_register", "source": "registering", "dest": None},
+            {
+                "trigger": "_trig_send_register_request",
+                "source": "initializing",
+                "before": "_send_register_request",
+                "dest": "registering",
+            },
+            {
+                "trigger": "_trig_wait",
+                "source": "registering",
+                "dest": "waiting_to_poll",
+                "after": "_wait_for_interval",
+            },
+            {
+                "trigger": "_trig_poll",
+                "source": "waiting_to_poll",
+                "dest": "polling",
+                "after": "_query_operation_status",
+            },
+            {
+                "trigger": "_trig_complete",
+                "source": "registering",
+                "dest": "completed",
+                "after": "_call_complete",
+            },
+            {
+                "trigger": "_trig_complete",
+                "source": "waiting_to_poll",
+                "dest": "completed",
+                "after": "_call_complete",
+            },
+            {
+                "trigger": "_trig_complete",
+                "source": "polling",
+                "dest": "completed",
+                "after": "_call_complete",
+            },
+            {
+                "trigger": "_trig_error",
+                "source": "registering",
+                "dest": "error",
+                "after": "_call_error",
+            },
+            {
+                "trigger": "_trig_error",
+                "source": "waiting_to_poll",
+                "dest": "error",
+                "after": "_call_error",
+            },
+            {
+                "trigger": "_trig_error",
+                "source": "polling",
+                "dest": "error",
+                "after": "_call_error",
+            },
+            {"trigger": "_trig_cancel", "source": "disconnected", "dest": None},
+            {
+                "trigger": "_trig_cancel",
+                "source": "registering",
+                "dest": "cancelling",
+                "after": "clear_and_disconnect",
+            },
+            {
+                "trigger": "_trig_cancel",
+                "source": "waiting_to_poll",
```

Comment:
>"source": "waiting_to_poll" [](start = 16, length = 27)

since you have 3 transitions that are identical except for the source state, you can use an array for the source and only have one transition.
e.g.:
{
    "trigger": "_trig_cancel",
    "source": [ "registering", "waiting_to_poll", "polling"],
    "dest": "cancelling",
    "after": "clear_and_disconnect",
},